compiler/src/iree/compiler/Codegen/Common/GPUPipelining.cpp

```diff
@@ -23,6 +25,49 @@ namespace iree_compiler {
 static const StringLiteral kPipeliningLoopMarker = "__pipelining_K_loop__";
 static const StringLiteral kPipeliningGlobalLoad = "__pipelining_global_load__";
 
+
+// Returns a new AsyncCopyOp with Zfill 
+static Operation* replaceAsyncCopywithAsyncCopyZfill(Operation* op, Value pred, PatternRewriter& rewriter) {
+
+  if (!isa<nvgpu::DeviceAsyncCopyOp>(op)) {
+    // Return the op if it is side effect free or barrier, commit group, or shared memory load
+    if (mlir::isSideEffectFree(op) || 
+        (isa<gpu::BarrierOp, nvgpu::DeviceAsyncCreateGroupOp, nvgpu::LdMatrixOp>(op)))
```

Comment:
I'm guessing we will getting other kind of ops for cases where ldmatrix cannot be used. Also we should make sure we don't use that path for cases where async.cp is not used. (the depth we pick in this case is 1 so peeling sounds like a good idea anyway)